connmgr.go

```diff
@@ -167,47 +169,89 @@ type peerInfo struct {
 // TODO: error return value so we can cleanly signal we are aborting because:
```

Comment:
```suggestion
This function blocks until a trim is completed. If a trim is underway, a new
one won't be started, and instead it'll wait until that one is completed before
returning.

// TODO: error return value so we can cleanly signal we are aborting because:
```